server/blob/blob-objectstorage/src/main/java/org/apache/james/blob/objectstorage/ObjectStorageBlobsDAO.java

```diff
@@ -102,40 +102,23 @@
             .contentLength(payload.getLength().orElse(new Long(data.length)))
             .build();
 
-        return save(bucketName, blob)
+        return Mono.fromRunnable(() -> blobPutter.putDirectly(blob))
```

Comment:
Don't we pass the bucketName as parameter ?

Reply:
We were not using it before

Reply:
Is it something we should create an issue for ?

Reply:
I'm already doing it, that is what I'm doing in my PRs. if this PR is merged soon I will fix this

Reply:
you use different blobId in two save method(), is that inconsistent?

Reply:
That's due to the nature of the data: one can have it's `BlobId` computed before reading it, the other after

Reply:
> Is it something we should create an issue for ?

That is somethid @trantienduchn is working on here: https://github.com/linagora/james-project/pull/2492 @remk